### PR TITLE
feat: support groups in a slice's primary section

### PR DESCRIPTION
**Pull request by angeloashmore**
> [!IMPORTANT]
> Before merging this PR, update `@prismicio/types-internal` once https://github.com/prismicio/prismic-types-internal/pull/70 is merged.

## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR adds support for group fields in a slice's primary section. This feature will be enabled in Slice Machine and Prismic at a future time.

No breaking changes are introduced in this PR.

**Related**: [DT-2078](https://linear.app/prismic/issue/DT-2078/aadevuser-i-can-use-prismicioclient-to-fetch-and-type-slice-content)

## Checklist:

- [ ] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

### CODE REVIEW

File: src/types/value/types.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import type { ContentRelationshipField } from "./contentRelationship";
 import type { DateField } from "./date";
 import type { EmbedField } from "./embed";
 import type { GeoPointField } from "./geoPoint";
+import { GroupField } from "./group";
```
Comment (lihbr) [inline src/types/value/types.ts:7]:
```suggestion
import type { GroupField } from "./group";
```

**Review by lihbr — APPROVED** [review]
LGTM, just a minor suggestion ☺️

File: src/types/value/types.ts (hunk 1 above)
Comment (angeloashmore) [reply src/types/value/types.ts:7]: Oops! Thanks, I saw this suggestion from your other review but forgot to apply it.

**Review by lihbr — APPROVED** [review]
_(just leaving that as a checkpoint for me, still looking good, waiting on internal types release)_

**Pull request merged**